lib/knigge/options.ex

```diff
@@ -305,7 +305,7 @@ defmodule Knigge.Options do
     do_not_delegate: :keyword,
     implementation: :module,
     otp_app: :atom,
-    config_key: :atom,
+    config_key: :keys,
```

Comment:
Instead of introducing `:keys` I'd argue it's a good opportunity to introduce an "or" mechanism:

```elixir
config_key: [:atom, {:list_of, :atom}]
```

Where `valid_value?/2` would look like this:

```elixir
defp valid_value?({:list_of, type}, value),
  do: is_list(value) and Enum.all?(value, &valid_value?(type, &1))

defp valid_value?(list, value) when is_list(list),
  do: Enum.any?(list, &valid_value?(&1, value))
```

In addition the `transform` function could then be extended to map `config_key` to always be a list (by using `List.wrap/1`).

Which in turn would simplify the implementation in `Implementation.fetch!/1` as there no need for multiple clauses anymore (one for single atom, one for list).